### PR TITLE
Simple fix to Issue 158 (and potentially other issues)

### PR DESCRIPTION
**Pull request by cgnorthcutt**
Fixes https://github.com/cleanlab/cleanlab/issues/158

A simple fix to clip the diagonal confident joint so that it avoids columns entirely of zero and always requires one correctly labeled example in every class.

* This is the lowest-level place where the confident joint is computed. The fix (should) trickle to all downstream use cases.
  - In your review, consider if this is the case.
* this PR reduces the warnings in our tests by half

### CODE REVIEW

File: docs/source/tutorials/tabular.ipynb [hunk 1]
```diff
@@ -391,7 +391,7 @@
    "source": [
     "# Hidden code cell to check that cleanlab has improved prediction accuracy\n",
     "if acc_og >= acc_cl:\n",
-    "    raise Exception(\"Cleanlab training failed to improve model accuracy.\")"
+    "    raise Exception(\"Cleanlab training failed to improve model accuracy.\")\n"
```
Comment (jwmueller) [inline docs/source/tutorials/tabular.ipynb:394]: this \n is causing notebook linter to fail.

**Review by jwmueller — APPROVED** [review]
LGTM

File: cleanlab/count.py [hunk 2]
```diff
@@ -410,6 +410,8 @@ def compute_confident_joint(
     true_labels_confident = true_label_guess[at_least_one_confident]
     labels_confident = labels[at_least_one_confident]
     confident_joint = confusion_matrix(true_labels_confident, labels_confident).T
+    # Guarantee at least one correctly labeled example is represented in every class
+    np.fill_diagonal(confident_joint, confident_joint.diagonal().clip(min=1))
```
Comment (jwmueller) [inline cleanlab/count.py:414]: Could consider printing warning if the results are likely  unreliable if there was a 0 before.  I do not know how to judge that.

File: docs/source/tutorials/tabular.ipynb (hunk 1 above)
Comment (anishathalye) [reply docs/source/tutorials/tabular.ipynb:394]: Why is there a change to this notebook at all?

**Comment by anishathalye** [comment]
I rebased onto c4e8462.

**Review by anishathalye — APPROVED** [review]
Summarizing discussion offline: decided to skip warning in this situation.

LGTM.

File: docs/source/tutorials/tabular.ipynb (hunk 1 above)
Comment (cgnorthcutt) [reply docs/source/tutorials/tabular.ipynb]: fixed

File: cleanlab/count.py (hunk 2 above)
Comment (cgnorthcutt) [reply cleanlab/count.py:416]: It shouldn't make a big difference. Will leave as is for now. we can re-evaluate if this seems to cause issues down the line.

**Pull request merged**